### PR TITLE
Draft: Add functional test

### PR DESCRIPTION
**Pull request by bryan-brancotte**

### CODE REVIEW

File: tests/test_functional_test.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import os
+
+from defense_finder_cli.main import run
+from tests import TooledTest, FakeExitCodeException
+
+
+def cleaned_lines(expected):
+    return set(sorted([line.upper() for line in expected.readlines()]))
```
Comment (jeanrjc) [inline tests/test_functional_test.py:8]: Why upper() ? I would remove it (tests don't currently pass because of this I think)

Comment (bryan-brancotte) [reply tests/test_functional_test.py:8]: When developing the tests, I saw some line where only the case differ, I though the case should not make the test fails but it is clearly open for debate ! `cleaned_lines` is used for both expected results and produced results, so using upper does not fails tests, it silences case issue.

Do you want the test to fail if the strain is written in different case ?

File: tests/run_tests.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+#                                                                       #
+# MacSyFinder is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of        #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          #
+# GNU General Public License for more details .                         #
+#                                                                       #
+# You should have received a copy of the GNU General Public License     #
+# along with MacSyFinder (COPYING).                                     #
+# If not, see <https://www.gnu.org/licenses/>.                          #
+#########################################################################
```
[the remark this answers is not in the document]
Comment (bryan-brancotte) [reply tests/run_tests.py:23]: It's the part where I don't know what to do as this file is a copy/paste from MacSyFinder. @bneron could you help me here ? What should I/we write ?

[the remark this answers is not in the document]
Comment (bneron) [reply tests/run_tests.py:23]: this part of code is a copy paste from macsyfinder. so you should not rewrite the license.

[the remark this answers is not in the document]
Comment (bryan-brancotte) [reply tests/run_tests.py]: Thanks for the feedback, I rewrote it

File: tests/test_functional_test.py (hunk 1 above)
Comment (bryan-brancotte) [reply tests/test_functional_test.py:8]: Hi @jeanrjc 
Do you want the test to be case sensitive or not ? For now they are not

**Comment by jeanrjc** [comment]
I modified the way get the model's version, the way you did it triggered the full macsyfinder's log. I also filter out other potential models that would be installed by the user

**Comment by jeanrjc** [comment]
There is a bug with coverage, I don't understand it. I have a different one on my laptop with coverage (`ImportError: cannot import name 'TooledTest' from 'tests'`). But the test passes with pytest.

The problem was 2 folds :
1. There was new systems or new names
2. The numbering of the system was shifted because of the new names. 

The 1 is important to capture, but not the 2 which add noise to the output and makes it harder to quickly pinpoint what is actually different. Maybe we could drop the column sys_id or at least strip the number at the end.

**Comment by bryan-brancotte** [comment]
Hi @jeanrjc 

I droppped the sys_id column in ed87285e58560b527c85c295b855fa8d4bb1e698, quicker than using regexp to strip then end (but doable)

**Comment by jeanrjc** [comment]
I thought this one was merged, but not ! Is there anything missing ?

**Comment by bryan-brancotte** [comment]
Hi, data in expected_results folders have to be updated so when the tests runs, results are the same. For the reste I think we are good (ignoring a specific column done, and case sensible comparison done to)

**Pull request merged**